tmt/steps/provision/vagrant.py

```diff
@@ -7,101 +7,267 @@
 from tmt.utils import ConvertError, SpecificationError, GeneralError, quote
 
-from click import echo
 from urllib.parse import urlparse
 
 
-# DATA[*]:
-#   HOW = libvirt|virtual|docker|container|vagrant|...
-#         provider, in Vagrant's terminilogy
-#
-#   IMAGE = URI|NAME
-#         NAME is for Vagrant or other HOW, passed directly
-#         URI can be path to BOX, QCOW2 or Vagrantfile f.e.
-#
-#   BOX = Set a BOX name directly (in case of URI for IMAGE)
-#
-class ProvisionVagrant(ProvisionBase):
-    """ Use Vagrant to Provision an environment for testing """
+class ProvisionVagrant(tmt.steps.provision.ProvisionPlugin):
+    """
+    Use Vagrant to Provision an environment for testing
+    """
+
+    # Guest instance
+    _guest = None
+
+    # Supported methods
+    _methods = [
+        tmt.steps.Method(name='virtual.vagrant', doc=__doc__, order=50),
+        tmt.steps.Method(name='libvirt.vagrant', doc=__doc__, order=50),
+        tmt.steps.Method(name='virtualbox.vagrant', doc=__doc__, order=50),
+        tmt.steps.Method(name='container.vagrant', doc=__doc__, order=50),
+        ]
+
+    display = ['image', 'box', 'memory', 'user', 'password', 'key', 'guest',
+        'vagrantfile', 'sync_type']
+
+    @classmethod
+    def options(cls, how=None):
+        """ Prepare command line options for testcloud """
+        return [
+            click.option(
+                '-i', '--image', metavar='IMAGE',
+                help='Select image to use. Short name or complete url.'),
+            click.option(
+                '-m', '--memory', metavar='MEMORY',
+                help='Set available memory in MB, 2048 MB by default.'),
+            click.option(
+                '-u', '--user', metavar='USER',
+                help='Username to use for all guest operations.'),
+            click.option(
+                '-b', '--box', metavar='BOX',
+                help='Explicit box name to use.'),
+            click.option(
+                '-p', '--password', metavar='PASSWORD',
+                help='Password for login into the guest system.'),
+            click.option(
+                '-k', '--key', metavar='KEY',
+                help='Private key for login into the guest system.'),
+            click.option(
+                '-g', '--guest', metavar='GUEST',
+                help='Select remote host to connect to (hostname or ip).'),
+            click.option(
+                '-f', '--vagrantfile', metavar='VAGRANTFILE',
+                help='Path to Vagrantfile, which will be copied.'),
+            click.option(
+                '-s', '--sync-type', metavar='SYNC_TYPE',
+                help='Sync method Vagrant will use.'),
+            ] + super().options(how)
+
+
+    def default(self, option, default=None):
+        """ Return default data for given option """
+        defaults = {
+            'user': 'root',
+            'memory': 2048,
+            'sync_type': 'rsync',
+            'image': 'fedora/32-cloud-base'
```

Comment:
We actually need to be specific with this (there's no way to set "latest"). But it may be generated from some other source (current stable Fedora number). Let me know if we have something like that available somewhere in the source (`util.py` f.e.).